### PR TITLE
[stable/postgresql] servicemonitor resource for prometheus operator

### PR DESCRIPTION
**Pull request by sebastien-prudhomme**
#### What this PR does / why we need it:

Add a servicemonitor resource for prometheus operator

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/postgresql/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    chart: {{ template "postgresql.chart" . }}
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+    {{- if .Values.metrics.serviceMonitor.additionalLabels }}
+{{ toYaml .Values.metrics.serviceMonitor.additionalLabels | indent 4 }}
+    {{- end }}
+spec:
+  endpoints:
+    - port: metrics
+      interval: 30s
```
Comment (juan131) [inline stable/postgresql/templates/servicemonitor.yaml]: This should be configurable. See https://github.com/helm/charts/blob/master/stable/mariadb/templates/servicemonitor.yaml#L24

Comment (juan131) [reply stable/postgresql/templates/servicemonitor.yaml]: *scrapeTimeout* should be configurable. See https://github.com/helm/charts/blob/master/stable/mariadb/templates/servicemonitor.yaml#L27

File: stable/postgresql/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{{- if and .Values.metrics.enabled .Values.metrics.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "postgresql.fullname" . }}
+  labels:
```
Comment (juan131) [inline stable/postgresql/templates/servicemonitor.yaml:9]: The namespace to deploy the ServiceMonitor should be configurable. See https://github.com/helm/charts/blob/master/stable/mariadb/templates/servicemonitor.yaml#L7

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the contribution! Please take a look to my comments

File: stable/postgresql/templates/servicemonitor.yaml (hunk 1 above)
Comment (sebastien-prudhomme) [reply stable/postgresql/templates/servicemonitor.yaml]: With default values or with default Prometheus values, as in these PR #14279

Comment (juan131) [reply stable/postgresql/templates/servicemonitor.yaml]: With default Prometheus values, as done at that PR

**Comment by juan131** [comment]
By the way, remember to rebase your branch from master to resolve the conflicts in the **Chart.yaml**

File: stable/postgresql/templates/servicemonitor.yaml (hunk 2 above)
Comment (sebastien-prudhomme) [reply stable/postgresql/templates/servicemonitor.yaml:9]: Done

File: stable/postgresql/templates/servicemonitor.yaml (hunk 1 above)
Comment (sebastien-prudhomme) [reply stable/postgresql/templates/servicemonitor.yaml]: Done

**Review by juan131 — APPROVED** [review]
Thanks so much for implementing the suggestions! Approving your changes.

**Pull request merged**